### PR TITLE
Add support for more authentication methods for the default calendar module

### PR DESCRIPTION
**Pull request by deg0nz**
HTTP Basic, Digest and OAuth2/Bearer authentications are now supported
by the calendar module

### CODE REVIEW

File: modules/default/calendar/calendar.js [hunk 1]
```diff
@@ -72,10 +72,18 @@ Module.register("calendar", {
 
 			var calendarConfig = {
 				maximumEntries: calendar.maximumEntries,
-				maximumNumberOfDays: calendar.maximumNumberOfDays,
+				maximumNumberOfDays: calendar.maximumNumberOfDays
 			};
 
-			this.addCalendar(calendar.url, calendar.user, calendar.pass, calendarConfig);
+			// we check user and password here for backwards compatibility with old configs
+			if(calendar.user && calendar.pass){
```
Comment (deg0nz) [inline modules/default/calendar/calendar.js]: I added this check for backwards compatibility, so that old configs don't break the calendar module

**Comment by deg0nz** [comment]
I tested Digest authentication against a Baïkal (sabre/dav) Server and OAuth/Bearer authentication against the Google calendar API. They're working fine.

(I didn't know how to get an .ics file out of Google's APIs, so `ical.js` couldn't parse the data, but the authentication mechanism with a BearerToken itself is working and valid data is returned on a request.)

Unfortunately I couldn't really test Basic auth, since I had no endpoint for this. But it should work as before, since it is used as default authentication if no auth method is given.

File: modules/default/calendar/README.md [hunk 2]
```diff
@@ -53,25 +53,36 @@ The `colored` property gives the option for an individual color for each calenda
 #### Default value:
 ````javascript
 config: {
-    	colored: false,
+    colored: false,
 	calendars: [
 		{
 			url: 'http://www.calendarlabs.com/templates/ical/US-Holidays.ics',
 			symbol: 'calendar',
+			auth: {
```
Comment (deg0nz) [inline modules/default/calendar/README.md]: I just discovered, that this example is at the wrong place...
I'll take care of this tomorrow.

**Comment by MichMich** [comment]
Since this is a large change: @roramirez, @fewieden, @nhubbard: does this change work for you guys?

**Comment by fewieden** [comment]
I don't have any calendars to test with :/

**Comment by deg0nz** [comment]
@fewieden I can give you a test account on my Baikal Server if you want. There you can test the digest auth.

For testing basic auth, I just had the idea that it suffices if one puts an `.ics` file on a webserver and specifies Basic authentication in the `.htaccess` file (on an apache server) for the folder. I can do that too and provide URL and test credentials for you.

Unfortunately I can't give you any test environment for Bearer auth, because I tested it against my private Google account. (You have to set up a google calendar API end point and generate an OAuth token to authenticate).

File: modules/default/calendar/calendar.js (hunk 1 above)
Comment (roramirez) [inline modules/default/calendar/calendar.js]: Here if there old configuration should show a warning like deprecated options.

File: modules/default/calendar/debug.js [hunk 3]
```diff
@@ -8,14 +8,22 @@
 
 var CalendarFetcher = require("./calendarfetcher.js");
 
-var url = "https://calendar.google.com/calendar/ical/pkm1t2uedjbp0uvq1o7oj1jouo%40group.calendar.google.com/private-08ba559f89eec70dd74bbd887d0a3598/basic.ics";
+var url = "https://calendar.google.com/calendar/ical/pkm1t2uedjbp0uvq1o7oj1jouo%40group.calendar.google.com/private-08ba559f89eec70dd74bbd887d0a3598/basic.ics";  // Standard test URL
+// var url = "https://www.googleapis.com/calendar/v3/calendars/primary/events/"; // URL for Bearer auth (must be configured  in Google OAuth2 first)
```
Comment (roramirez) [inline modules/default/calendar/debug.js]: I dont know why should change this.

**Comment by roramirez** [comment]
I've test and create a test cases for testsuite and this change pass all configurations with basic-auth.

if necessary add more test case for _digest_ and _bearer_  there a samples on Pull Request #768 can anyone can take.

For me can aprove this change after resolved some little things exposed in my review.

File: modules/default/calendar/debug.js (hunk 3 above)
Comment (deg0nz) [inline modules/default/calendar/debug.js]: Yeah, you're right. Me neither ^^'

**Comment by deg0nz** [comment]
Yes, I'm gonna add the tests. At least one for digest auth.

File: modules/default/calendar/calendar.js (hunk 1 above)
Comment (deg0nz) [inline modules/default/calendar/calendar.js]: I don't really understand this. How and where should deprecation options appear?

Comment (fewieden) [inline modules/default/calendar/calendar.js]: If someone is using the old configuration sytax add a warning/info to the console

Comment (deg0nz) [inline modules/default/calendar/calendar.js]: OK, I want to do this right. Should I just add something like: `Log.warn("Deprecation warning: Please update your calendar authentication configuration.")`?

Or should it contain more information?

Comment (nhubbard) [inline modules/default/calendar/calendar.js]: @deg0nz Looks good.

**Comment by nhubbard** [comment]
@MichMich If @roramirez is working on tests, then that's fine... but we need to fix the merge conflict in the calendar README.

**Comment by deg0nz** [comment]
@nhubbard Before you fix the merge conflict, please let me fix the example I added for the auth options. I accidentally added it to the default example, which doesn't need any authentication at all... 
Or should I leave it there?

**Comment by deg0nz** [comment]
Hey guys, just FYI:
I don't really have time to get the fixes done this week, I'm planning to get all of this done by the weekend (fixes of your reviews and the additional tests in #768 ).
Sorry for the delay, I hope it's OK for you guys.

**Comment by roramirez** [comment]
Hi @deg0nz:
Do you still interested in that you mentioned before?

**Comment by deg0nz** [comment]
Hey @roramirez 
Yes, I am. I was just really confused that @MichMich already merged the pull requests... And I didn't know what to do.
Also, I had not so much free time in the past days :/

So how should this work? I just make the changes and make a new PR?

**Comment by roramirez** [comment]
Yes, only need a new Pull Request with changes.

**Comment by deg0nz** [comment]
okay @roramirez 
At first, I want to read into Mocha a little bit and the I'll create tests. Unfortunately my free time is a little bit sparse at the moment, but I'll get the job done in some time ;)

**Pull request merged**